components/brave_wallet/browser/blockchain_registry_unittest.cc

```diff
@@ -460,21 +462,211 @@ TEST(BlockchainRegistryUnitTest, GetProvidersBuyTokens) {
   base::test::TaskEnvironment task_environment;
   auto* registry = BlockchainRegistry::GetInstance();
 
-  // Get Wyre, Ramp, Sardine and Transak buy tokens.
-  base::RunLoop run_loop;
+  std::vector<mojom::BlockchainToken> buy_tokens;
```

Comment:
```
TEST(BlockchainRegistryUnitTest, GetProvidersBuyTokens) {
  base::test::TaskEnvironment task_environment;
  auto* registry = BlockchainRegistry::GetInstance();

  std::vector<mojom::BlockchainToken> buy_tokens;
  for (const auto& v : {GetWyreBuyTokens(), GetRampBuyTokens(),
                        GetSardineBuyTokens(), GetTransakBuyTokens()}) {
    buy_tokens.insert(buy_tokens.end(), v.begin(), v.end());
  }

  const char* chains[] = {
      mojom::kMainnetChainId,          mojom::kPolygonMainnetChainId,
      mojom::kAvalancheMainnetChainId, mojom::kBinanceSmartChainMainnetChainId,
      mojom::kSolanaMainnet,           mojom::kCeloMainnetChainId,
      mojom::kArbitrumMainnetChainId};

  for (auto* chain : chains) {
    std::vector<mojom::BlockchainTokenPtr> expected_tokens;
    for (const auto& token : buy_tokens) {
      if (token.chain_id == chain) {
        expected_tokens.push_back(mojom::BlockchainToken::New(token));
      }
    }

    base::RunLoop run_loop;
    registry->GetProvidersBuyTokens(
        {mojom::OnRampProvider::kWyre, mojom::OnRampProvider::kRamp,
         mojom::OnRampProvider::kSardine, mojom::OnRampProvider::kTransak,
         mojom::OnRampProvider::kTransak /* test duplicate provider */},
        chain,
        base::BindLambdaForTesting(
            [&](std::vector<mojom::BlockchainTokenPtr> token_list) {
              EXPECT_NE(token_list.size(), 0UL);
              EXPECT_NE(expected_tokens.size(), 0UL);
              EXPECT_TRUE(std::equal(expected_tokens.begin(),
                                     expected_tokens.end(), token_list.begin(),
                                     token_list.end()))
                  << chain;

              run_loop.Quit();
            }));
    run_loop.Run();
  }
}
```

Reply:
@simoarpe we can just write
`EXPECT_EQ(expected_tokens, token_list) << chain;`
no need to check both sizes and std::equal.